pkg/provisioner/ironic/hardwaredetails/hardwaredetails.go

```diff
@@ -59,24 +59,35 @@ func getNICSpeedGbps(intfExtradata introspection.ExtraHardwareData) (speedGbps i
 func getNICDetails(ifdata []introspection.InterfaceType,
 	basedata map[string]introspection.BaseInterfaceType,
 	extradata introspection.ExtraHardwareDataSection) []metal3v1alpha1.NIC {
-	nics := make([]metal3v1alpha1.NIC, len(ifdata))
-	for i, intf := range ifdata {
+	var nics []metal3v1alpha1.NIC
+	for _, intf := range ifdata {
 		baseIntf := basedata[intf.Name]
 		vlans, vlanid := getVLANs(baseIntf)
-		ip := intf.IPV4Address
-		if ip == "" {
-			ip = intf.IPV6Address
+		if intf.IPV4Address != "" {
```

Comment:
Is it possible for the list of interfaces to include any that do not have addresses? Under the old approach, those would show up in the list without an IP. Now they will not appear in the list at all.

Reply:
Ok updated to handle this case, and added a test.